src/command_modules/azure-cli-container/azure/cli/command_modules/container/custom.py

```diff
@@ -104,6 +105,15 @@ def create_container(cmd,
 
     container_resource_requirements = _create_resource_requirements(cpu=cpu, memory=memory)
 
+    if (len(image.split('/')) > 1) and image.split('/')[0].endswith('.azurecr.io'):
```

Comment:
This won't work with other cloud.
/cc @tjprescott @sajayantony This discussion happened long time ago, do you think we can expose login server suffix here https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/cloud.py#L148 and for each cloud?